internal/federation/server.go

```diff
@@ -30,9 +30,14 @@ import (
 	"github.com/matrix-org/util"
 
 	"github.com/matrix-org/complement/internal/config"
-	"github.com/matrix-org/complement/internal/docker"
 )
 
+// Subset of Deployment used in federation
+type FederationDeployment interface {
+	GetConfig() *config.Complement
+	RoundTripper() http.RoundTripper
+}
```

Comment:
So a FederationDeployment is a Deployment, but not the other way round?

I'm not sure that makes much sense, since one could get two regular Deployments to federate with each other?

Is the point that internal/federation represents a complement-controlled mock homeserver?

Reply:
generally speaking if I have a universe of objects Foo and some adjective or noun X, I would expect an XFoo to be a more specific interface rather than less specific. (Unless X is `Base` or `Abstract`, I suppose.)

Reply:
`Deployment` is a collection of running homeservers which are federating with each other. There can just be 1 homeserver in which case it isn't federating. It exposes a bunch of functions for interacting with these servers.

`FederationDeployment` is a subset of `Deployment` functions which are only relevant for the internal complement server. This is mostly out of neccessity rather than anything else, as if I passed in a `Deployment` I'd get a circular reference loop. In essence, `FederationDeployment` is just the bits of `Deployment` this file needs to work correctly.